### ISSUE TITLE
Config settings are wrong and cause NIS not to show up after flashing

### ISSUE THREAD

**Issue by ideas32** [FYSETC/Nozzle-Input-Shaper#2]
After flashing my NIS by following this guide, it wouldn't show up as a USB device. I then experimented a bit more, I figured out that instead of using ```PA11/PA12``` for USB, I had to use ```PA9/PA10``` and ```Internal clock``` instead of ```8 MHz crystal```. These options were hidden unless I selected ```Enable extra low-level configuration options```, which is why I think they were not recommended in the guide, even though the schematic shows USB is on ```PA9/PA10```. After compiling with this configuration, the NIS showed up and worked properly.

**Comment by Dnic94** [comment]
+1, I had the same problem. This config works.
@ideas32 Thanks for debugging.

**Comment by thousandmoths** [comment]
tried your config and still not detected after flash.  Will play around a bit more

**Comment by DeadiPodFTW** [comment]
I think I solved this issue.

One has to make sure ALL MCU's are recompiled and flashed to updated versions before proceeding with the instructions (including the changes from ideas32 (good work btw and thank you ideas32)).

I hope this helps someone, as I was lost for so long. Then got a "tip" from my Klipper host telling me this exactly.

**Comment by Dr-Info** [comment]
After going through and trying to flash this NIS I ran into several issues. Make sure all of your MCU's have the latest klipper version. Below should be the correct instructions based on the wonderful help from @ideas32

https://github.com/Dr-Info/Nozzle-Input-Shaper/blob/main/README.md

**Comment by devolf92** [comment]
Same issue, i decided to update all klipper device firmware, and after flashing with official guide, i received a brick. 
thanks for advice, this config works fine
![image](https://github.com/FYSETC/Nozzle-Input-Shaper/assets/15054594/89c39044-bc0c-44f7-ab79-54e4b19935e2)

**Comment by ideas32** [comment]
> tried your config and still not detected after flash.  Will play around a bit more
> 
> 

Any updates?

**Comment by Klaiii** [comment]
> After going through and trying to flash this NIS I ran into several issues. Make sure all of your MCU's have the latest klipper version. Below should be the correct instructions based on the wonderful help from @ideas32
> 
> https://github.com/Dr-Info/Nozzle-Input-Shaper/blob/main/README.md
> 

I followed you guide and i have to thank you so much! I was going Crazy.

It isnt specified at point 7 to do It in DFU Mode and i modified my printer.cfg from `[mcu NIS]` tò `[mcu USB]`.

**Comment by Dr-Info** [comment]
> I followed you guide and i have to thank you so much! I was going Crazy.
> 
> It isnt specified ad point 7 to do It in DFU Mode and i modified my printer.cfg from `[mcu NIS]` tò `[mcu USB]`.

Yeah I discovered the same thing when I was trying to flash my device. Light bulb moment went off and I was like it's just another device with klipper so just follow the normal ops for other MCU's just different make menuconfig. I think the only way is to flash it via DFU mode. The steps above were just compiling the firmware for flashing. Yeah you can name the config file and MCU whatever you want for the IS. 

Thanks for the feedback

**Comment by ideas32** [comment]
Looks like FYSETC has fixed their config and instructions so I'm going to close this issue.
